agents/hwp_gripper/gripper_agent.py

```diff
@@ -0,0 +1,575 @@
+                            'Actuator 2 Warm', 'Actuator 3 Cold', 'Actuator 3 Warm']
+        self.limit_pru = [8, 9, 10, 11, 12, 13]
+        self.limit_state = [0, 0, 0, 0, 0, 0]
+
+        self.mode = multiprocessing.Value(ctypes.c_bool, False)
+        self.force = multiprocessing.Value(ctypes.c_bool, False)
+
+        agg_params = {'frame_length': 60}
+        self.agent.register_feed('hwpgripper', record=True, agg_params=agg_params)
+        self.agent.register_feed('gripper_action', record=True)
+
+    @ocs_agent.param('auto_acquire', default=True, type=bool)
+    def init_processes(self, session, params):
+        if self._initialized:
+            self.log.info('Connection already initialized. Returning...')
+            return True, 'Connection already initialized'
+
+        self.client = gclient.GripperClient(self.mcu_ip, self.control_port, self.return_port)
+        self.collector = gc.GripperCollector(self.pru_port)
+
+        self.agent.start('grip_collect_pru')
+        self.agent.start('grip_build_pru')
+        # self.agent.start('grip_monitor')
+
+        if params['auto_acquire']:
+            self.agent.start('grip_acq')
+
+        self._initialized = True
+        return True, 'Processes started'
+
+    def grip_on(self, session, params=None):
+        with self.lock.acquire_timeout(0, job='grip_on') as acquired:
+            if not acquired:
+                self.log.warn('Could not perform action because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            if self.shutdown_mode:
+                return False, 'Shutdown mode is in effect'
+
+            self.log.info(self._send_command('ON'))
+
+        return True, 'Power on'
+
+    def grip_off(self, session, params=None):
+        with self.lock.acquire_timeout(0, job='grip_off') as acquired:
+            if not acquired:
+                self.log.warn('Could not perform action because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            if self.shutdown_mode:
+                return False, 'Shutdown mode is in effect'
+
+            self.log.info(self._send_command('OFF'))
+
+        return True, 'Power off'
+
+    @ocs_agent.param('state', default='ON', type=str)
+    @ocs_agent.param('actuator', default=0, type=int, check=lambda x: 0 <= x <= 3)
+    def grip_brake(self, session, params=None):
+        with self.lock.acquire_timeout(0, job='grip_brake') as acquired:
+            if not acquired:
+                self.log.warn('Could not perform action because {} is already running'.format(self.lock.job))
+                return False, 'Could not acquire lock'
+
+            if self.shutdown_mode:
+                return False, 'Shutdown mode is in effect'
+
+            if params['actuator'] == 0:
+                self.log.info(self._send_command('BRAKE ' + params['state']))
```

Comment:
The response will depend on the command given, but in all cases it will be a Boolean value. For "read" commands (ALARM, INP, ACT, RESET) the Boolean value is a response to the specific query (e.i. if the ALARM command returns True, then there is an alarm. If it instead returns False, then there is no alarm). For "write" commands (POWER, HOME, MOVE, BRAKE, EMG) the Boolean value is whether the action was carried out successfully (e.i. if the HOME command returns True, then the actuators were homed without issue. If it instead returns False, then there was some issue preventing the action from being completed).